broker-core/src/main/java/io/zeebe/broker/subscription/message/state/MessageSubscriptionState.java

```diff
@@ -125,27 +132,56 @@ private int writeMessageNameAndCorrelationKey(
   }
 
   private int writeSentTimeKey(MutableDirectBuffer buffer, final MessageSubscription subscription) {
+    final int expectedLength = Long.BYTES + getSubscriptionKeyLength(subscription);
     int offset = 0;
 
     buffer.putLong(offset, subscription.getCommandSentTime(), STATE_BYTE_ORDER);
     offset += Long.BYTES;
 
+    offset = writeSubscriptionKey(buffer, offset, subscription);
+    assert offset == expectedLength : "End offset differs from expected length";
+
+    return offset;
+  }
+
+  private int writeSubscriptionKey(
+      MutableDirectBuffer buffer, int offset, final MessageSubscription subscription) {
+    final DirectBuffer messageName = subscription.getMessageName();
+    final int startOffset = offset;
+    final int expectedLength = getSubscriptionKeyLength(subscription);
+
     buffer.putLong(offset, subscription.getElementInstanceKey(), STATE_BYTE_ORDER);
     offset += Long.BYTES;
 
-    assert (2 * Long.BYTES) == offset
-        : "Offset problem: offset is not equal to expected key length";
+    offset = writeIntoBuffer(buffer, offset, messageName);
+    assert (offset - startOffset) == expectedLength : "End offset differs from expected length";
+
     return offset;
   }
 
-  private boolean readSubscription(long key, MessageSubscription subscription) {
+  private void wrapSubscriptionKey(
+      DirectBuffer source, int offset, final MessageSubscription subscription) {
+    subscription.setElementInstanceKey(source.getLong(offset, STATE_BYTE_ORDER));
+    offset += Long.BYTES;
+
+    readIntoBuffer(source, offset, subscription.getMessageName());
+  }
+
+  private int getSubscriptionKeyLength(final MessageSubscription subscription) {
+    return Long.BYTES + Integer.BYTES + subscription.getMessageName().capacity();
+  }
+
+  private boolean readSubscription(MessageSubscription subscription) {
```

Comment:
It looks confusing to me that you use the properties of the subscription to find the persisted subscription in the state and also to write the persistent state into it. I would prefer separate method arguments.